examples/splinterdb_custom_ipv4_addr_sortcmp_example.c

```diff
@@ -0,0 +1,395 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * ----------------------------------------------------------------------------
+ * SplinterDB Advanced Iterators Example Program with custom sort-comparison.
+ *
+ * What's new beyond previous splinterdb_iterators_example.c?
+ *
+ * In this program, we show the application of user-specified custom
+ * key-comparison routines. The 'key' here is the 4-part IP-address, which
+ * is stored as the string seen from 'ping'; i.e. "208.80.154.232" .
+ * To illustrate the use of user-defined keys, we then provide a
+ * sort-comparison routine, which splits up the IP-address to its
+ * constituent parts, and does a numeric comparison of each 1-byte value.
+ *
+ * See:
+ * - The definition of custom splinter_data_cfg.key_compare to the
+ *   user-provided comparison function, custom_key_compare()
+ * - The ip4_ipaddr_keycmp() and ip4_split() functions that show how one
+ *   can deal with application-specific key formats.
+ * ----------------------------------------------------------------------------
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "splinterdb/default_data_config.h"
+#include "splinterdb/splinterdb.h"
+
+/*
+ * App-specific 'defaults' that can be parameterized, eventually.
+ */
+#define DB_FILE_NAME    "splinterdb_custom_ipv4_sortcmp_example_db"
+#define DB_FILE_SIZE_MB 1024 // Size of SplinterDB device; Fixed when created
+#define CACHE_SIZE_MB   64   // Size of cache; can be changed across boots
+
+// Describe the layout of fields in an IP4-address
+#define APP_IPV4_NUM_FIELDS 4
+#define APP_IPV4_NUM_DOTS   (APP_IPV4_NUM_FIELDS - 1)
+
+// Application declares the limit of key-sizes it intends to use
+#define APP_MAX_KEY_SIZE ((3 * APP_IPV4_NUM_FIELDS) + APP_IPV4_NUM_DOTS)
+
+// Max # of chars in a well-formed IP4 address, including null-terminator byte
+#define APP_IPV4_MAX_KEY_BUF_SIZE (APP_MAX_KEY_SIZE + 1)
```

Comment:
I would get rid of the `APP_` prefix

Reply:
Done. Replaced all mnemonics consistently with `IP4_` prefix.